ERCS/erc-7254.md

```diff
@@ -0,0 +1,141 @@
+
+Revenue Sharing is an extended version of [ERC-20](./eip-20.md). It proposes an additional payment method for token holders. 
+
+This standard includes updating rewards for holders and allowing token holders to withdraw rewards.
+
+## Motivation
+
+Potential use cases encompass:
+
+* Companies distributing dividends to token holders.
+* Direct sharing of revenue derived from business activities, such as marketplaces, Automated Market Makers (AMMs), and games.
+
+## Specification
+
+### Structures
+
+#### `UserInformation`
+
+`inReward`: when user's balance decreases, `inReward` will be updated
+`outReward`: when user's balance increases, `outReward` will be updated
+`withdraw`: total amount of reward tokens withdrawn
+
+```solidity
+struct UserInformation {
+    uint256 inReward;
+    uint256 outReward;
+    uint256 withdraw;
+}
+```
+
+### Methods
+
+#### `maxTokenReward`
+
+Returns max token reward.
+
+```solidity
+function maxTokenReward() public view returns (uint256)
+```
+
+#### `informationOf`
+
+Returns the account information of another account with the address `token` and `account`, including: `inReward`, `outReward` and `withdraw`.
+
+```solidity
+function informationOf(address token, address account) public view returns (UserInformation memory)
+```
+
+#### `informationOfBatch`
+
+Returns the list account information of another account with the `account`, including: `inReward`, `outReward` and `withdraw`.
+
+```solidity
+function informationOfBatch(address account) public view returns (UserInformation[] memory)
+```
+
+#### `tokenReward`
+
+Returns the list token reward address of the token.
+
+```solidity
+function tokenReward() public view returns (address[] memory)
+```
+
+#### `updateReward`
+
+Updates `rewardPerShare` of token reward.
+
+```
+rewardPerShare = rewardPerShare + amount / totalSupply()
+```
+
+```solidity
+function updateReward(address[] memory token, uint256[] memory amount) public
+```
+
+#### `viewReward`
+
+Returns the list amount of reward for an account
+
+```solidity
+function viewReward(address account) public view returns (uint256[] memory)
+```
+
+#### `getReward`
+
+Gets and returns reward with list token reward.
+
+```solidity
+function getReward(address[] memory token) public
+```
+
+#### `getRewardPerShare`
+
+Returns the reward per share of token reward.
+
+```solidity
+function getRewardPerShare(address token) public view returns (uint256)
+```
+
+#### `existsTokenReward`
+
+Returns the status of token reward.
+
+```solidity
+function existsTokenReward(address token) public view returns (bool)
+```
+
+## Rationale
+
+<!-- TODO -->
+
+## Reference Implementation
+
+* [`ERC7254`](../assets/eip-7254/ERC7254.sol)
+* [`IERC7254`](../assets/eip-7254/IERC7254.sol)
+
+## Security Considerations
+
+<!-- TODO -->
+
+Needs discussion.
```

Comment:
Ah, I didn't realize this was going into Review. You'll need to complete the Rationale and Security Considerations sections before you can merge this.